backend/src/schema/resolvers/follow.spec.js

```diff
@@ -40,84 +97,73 @@ afterEach(async () => {
 describe('follow', () => {
   describe('follow user', () => {
     describe('unauthenticated follow', () => {
-      it('throws authorization error', async () => {
-        const client = new GraphQLClient(host)
-        await expect(client.request(mutationFollowUser('u2'))).rejects.toThrow('Not Authorised')
+      test('throws authorization error', async () => {
+        authenticatedUser = null
+        const { errors, data } = await mutate({
+          mutation: mutationFollowUser,
+          variables,
+        })
+        expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
+        expect(data).toMatchObject({ follow: null })
       })
     })
 
-    it('I can follow another user', async () => {
-      const res = await clientUser1.request(mutationFollowUser('u2'))
-      const expected = {
-        follow: true,
-      }
-      expect(res).toMatchObject(expected)
-
-      const { User } = await clientUser1.request(`{
-        User(id: "u2") {
-          followedBy { id }
-          followedByCurrentUser
-        }
-      }`)
-      const expected2 = {
-        followedBy: [{ id: 'u1' }],
+    test('I can follow another user', async () => {
+      const { data: result } = await mutate({
+        mutation: mutationFollowUser,
+        variables,
+      })
+      const expectedUser = {
+        name: user2.name,
+        followedBy: [{ id: user1.id, name: user1.name }],
         followedByCurrentUser: true,
       }
-      expect(User[0]).toMatchObject(expected2)
+      expect(result).toMatchObject({ follow: expectedUser })
     })
 
-    it('I can`t follow myself', async () => {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-16T19:45:27Z" title="Monday, September 16th 2019, 9:45:27 pm +02:00">Sep 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -87,32 +129,39 @@ describe('follow', () => {
     })
   })
   describe('unfollow user', () => {
+    beforeEach(async () => {
+      await mutate({ mutation: mutationFollowUser(user2.id) })
```


You can either setup the data through a mutation or you can do:

```js
await myUser.relateTo(anotherUser, 'following')
```

If `myUser` and `anotherUser` is a neode node (that's the return type of the factory method).

Reply:
> [<img alt="vbelolapotkov" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/vbelolapotkov) **Authored by [vbelolapotkov](https://github.com/vbelolapotkov)**
_<time datetime="2019-09-17T06:55:22Z" title="Tuesday, September 17th 2019, 8:55:22 am +02:00">Sep 17, 2019</time>_

---


@roschaefer I wouldn't agree on this. Suggested improvement manipulates DB "directly" rather than through known server API, but we test API and we shouldn't reason about the DB or any other tools used by the server.

In the future relation name, `neode` model or some other properties can change which will break the test setup. So we'll have to get back and fix the test.

If we keep the current approach the test setup will break only if we change mutation definition. However, if this happens we'll have to rewrite all the test anyways.